rskj-core/src/main/java/co/rsk/RskContext.java

```diff
@@ -1024,6 +1027,10 @@ public synchronized List<InternalService> buildInternalServices() {
             internalServices.add(getPeerScoringReporterService());
         }
 
+        if (getRskSystemProperties().isRestServerEnabled()) {
```

Comment:
About this logic, the RestServer it is not a "Service" itself. The thing providing the service or the feature is in this case the Health check module. With this logic we could have active a rest server without any real feature using it. 

I would update how this service is handled make the health check module the the one to start the rest server if needed.

Reply:
The idea behind this `REST` server is for it to be a standalone server, for now, the only module will be `HealthCheck` one, but in the future, if we have to add new a new `REST` module, we'll simply have to add a new module and that's it.

Modules can be configured independently, things like status (active/inactive) and or different things can be tweaked via configuration file.

Modules don't have the responsibility to start the `Rest` server, but the other way around.

What we can do, thinking about responsibilities, is to add an "auto shutdown feature" if no modules are active (why we would have a running and listening server is all modules are inactive?).